File: app/src/main/java/com/tencent/util/LoggerUtils.java
package com.tencent.util;

import android.util.Log;

public class LoggerUtils {
	
	private static boolean isOpen = true;
	private static String TAG = "LoggerUtils";

	public static void i(String msg){
		if(isOpen && msg!= null){
			Log.i(TAG, msg);
		}
	}
	public static void d(String msg){
		if(isOpen&& msg!= null){
			Log.d(TAG, msg);
		}
	}

		public static void d(String tag,String msg){
		if(isOpen&& msg!= null){
			Log.d(TAG, tag + "---->" + msg);

		}
	}


	public static void e(String msg){
		if(isOpen&& msg!= null){
			Log.e(TAG, msg);
		}
	}
	public static void format_debug(String format,Object... args){
		if(isOpen&& args!= null && format != null){
			Log.d(TAG, String.format(format,args));
		}
	}
	public static void setDebug(boolean debug){
		isOpen = debug;
	}
}

File: app/src/main/java/com/booyue/utils/LoggerUtils.kt
package com.booyue.utils

import android.util.Log

/**
 * Created by Tianluhua on 2018\11\8 0008.
 */
class LoggerUtils {
    companion object {
        private var isOpen = true
        private val TAG = "LoggerUtils"

        fun i(msg: String) {
            if (isOpen) {
                Log.i(TAG, msg)
            }
        }

        fun d(msg: String) {
            if (isOpen) {
                Log.d(TAG, msg)
            }
        }

        fun d(msg: String, tag: String) {
            if (isOpen) {
                Log.d(TAG, "$tag ----->" + msg)
            }
        }

        fun e(msg: String) {
            if (isOpen) {
                Log.e(TAG, msg)
            }
        }

        fun e(msg: String, tag: String) {
            if (isOpen) {
                Log.e(TAG, "$tag ----->" + msg)
            }
        }

        fun setDebug(debug: Boolean) {
            this.isOpen = debug
        }

        fun format_debug(format: String, vararg args: Any) {
            if (isOpen) {
                Log.e(TAG, String.format(format, args))
            }
        }
    }
}